com.unity.sg2/Editor/GraphDeltaRegistry/Registry/SamplerType.cs

```diff
@@ -0,0 +1,172 @@
+
+        private static string GetUniqueSamplerName(FieldHandler field)
+        {
+            return field.ID.FullPath.Replace(".", "_");
+        }
+
+        public enum Filter { Point, Linear, Trilinear }
+        public enum Wrap { Clamp, Repeat, Mirror, MirrorOnce } // optionally can be per component- can be added later.
+        public enum Aniso { None = 0, Ansio2 = 2, Ansio8 = 8, Ansio16 = 16 } // optional
+
+        #region GetSet
+        public static void SetDepthComparison(FieldHandler field, bool enable)
+        {
+            try     { field.GetSubField<bool>(kCompare).SetData(enable); }
+            catch   { field.AddSubField<bool>(kCompare, enable); }
+        }
+        public static bool GetDepthComparison(FieldHandler field)
+        {
+            try     { return field.GetSubField<bool>(kCompare).GetData(); }
+            catch   { return false;  }
+        }
+        public static void SetAniso(FieldHandler field, Aniso aniso)
+        {
+            try     { field.GetSubField<Aniso>(kAniso).SetData(aniso); }
+            catch   { field.AddSubField<Aniso>(kAniso, aniso); }
+        }
+        public static Aniso GetAniso(FieldHandler field)
+        {
+            try     { return field.GetSubField<Aniso>(kAniso).GetData(); }
+            catch   { return Aniso.None; }
+
+        }
+        public static void SetFilter(FieldHandler field, Filter filter)
+        {
+            try     { field.GetSubField<Filter>(kFilter).SetData(filter); }
+            catch   { field.AddSubField<Filter>(kFilter, filter); }
+        }
+        public static Filter GetFilter(FieldHandler field)
+        {
+            try     { return field.GetSubField<Filter>(kFilter).GetData(); }
+            catch   { return Filter.Linear; }
+        }
+
+        public static void SetWrap(FieldHandler field, Wrap wrap)
+        {
+            try     { field.GetSubField<Wrap>(kWrap).SetData(wrap); }
+            catch   { field.AddSubField<Wrap>(kWrap, wrap); }
+        }
+        public static Wrap GetWrap(FieldHandler field)
+        {
+            try     { return field.GetSubField<Wrap>(kWrap).GetData(); }
+            catch   { return Wrap.Repeat; }
+        }
+
+        public static bool IsInitialized(FieldHandler field)
+        {
+            return field.GetSubField<Filter>(kFilter) != null
+                || field.GetSubField<Wrap>(kWrap) != null
+                || field.GetSubField<Aniso>(kAniso) != null
+                || field.GetSubField<bool>(kCompare) != null;
+        }
+
+
+        #endregion
+
+
+        public void BuildType(FieldHandler field, Registry registry)
+        {
+        }
+
+        public ShaderType GetShaderType(FieldHandler field, ShaderContainer container, Registry registry)
+        {
+            return container._UnitySamplerState;
+        }
+
+        public string GetInitializerList(FieldHandler field, Registry registry)
+        {
+            return $"In.{GetUniqueSamplerName(field)}";
+        }
+
```

Comment:
Any inline value that is promoted to material property eventually becomes a part of the block input structure, which is always called `In.` So the initial value for a SamplerState reference is going to come from the In struct.